dask/array/tests/test_array_core.py

```diff
@@ -2286,6 +2286,24 @@ def test_astype():
     assert d.astype("f8") is d
 
 
+def test_astype_gh1151():
```

Comment:
FYI, I did not write this test (it looks like Matt Rocklin did 6 years ago). But I want to group all the tests for `astype` together`, so I've moved it to go next to the others.